### PR TITLE
feat: fetch user info via js

### PR DESCRIPTION
**Pull request by Oakchris1955**
Before merging:
- edit `about/userinfo.json` to have your info too

### CODE REVIEW

**Review by 8u83r70 — APPROVED** [review]
Cum

**Comment by Oakchris1955** [comment]
To add yourself to the website, go to `assets/userinfo.json` and add these according to you under `Users`
@LukasDoesDev ,check this out
```json
"_username goes here_": {
	"FirstName": "_first name goes here_",
	"UserIcon": "_the url to your icon goes here_",
	"About": "_add anything you want here_",
	"Connections": {
		"Github": "_add connections here. key is connection name, value is URL/JS code_",
		"Discord": "_you can input as many as you want. if yours doesn't exist, just add it to Icons_"
	}
}
```

File: about/index.html [hunk 1]
```diff
@@ -33,159 +29,14 @@ <h3>
...
-                        <img src="assets/github.svg">
-                    </a>
-                </div>
-            </div>
-        </div>
-    </div>
-    <div id="footer">
-        <p>&gt;2022&lt;<br>SBD</p>
+
+        <div id="gallery"></div>
```
Comment (splatterxl) [inline about/index.html:33]: make this a semantic `<article>`

File: about/index.html [hunk 2]
```diff
@@ -33,159 +29,14 @@ <h3>
...
-    <div id="footer">
-        <p>&gt;2022&lt;<br>SBD</p>
+
+        <div id="gallery"></div>    
+    	</div>
+    <div w3-include-html='/general/footer.html' id="footer">
+
     </div>
+
+	<script>includeHTML()</script>
```
Comment (splatterxl) [inline about/index.html:39]: don't have this in a function, make it run when the script is started

File: about/userinfo.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (splatterxl) [inline about/userinfo.json:1]: make these properties snake_case or at least camelCase not PascalCase

File: general/bar.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<a class="item" href="">
```
Comment (splatterxl) [inline general/bar.html:1]: can you rename this folder to "_include"

File: init.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+function includeHTML() {
```
Comment (splatterxl) [inline init.js:1]: ideally this wouldn't be js for older browsers & non-js users

File: init.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	var z, i, elmnt, file, xhttp;
+	/*loop through a collection of all HTML elements:*/
+	z = document.getElementsByTagName("*");
+	for (i = 0; i < z.length; i++) {
+		elmnt = z[i];
+		/*search for elements with a certain atrribute:*/
+		file = elmnt.getAttribute("w3-include-html");
+		if (file) {
+			/*make an HTTP request using the attribute value as the file name:*/
+			xhttp = new XMLHttpRequest();
```
Comment (splatterxl) [inline init.js:11]: use fetch

File: init.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+			xhttp.onreadystatechange = function () {
+				if (this.readyState == 4) {
+					if (this.status == 200) { elmnt.innerHTML = this.responseText; }
+					if (this.status == 404) { elmnt.innerHTML = "Error while loading."; }
```
Comment (splatterxl) [inline init.js:15]: use textContent instead of innerHTML

File: init.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+					/*remove the attribute, and call this function once more:*/
+					elmnt.removeAttribute("w3-include-html");
+					includeHTML();
```
Comment (splatterxl) [inline init.js:18]: why are you recursively calling

File: init.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+				}
+			}
+			xhttp.open("GET", file, true);
+			xhttp.send();
+			/*exit the function:*/
+			return;
```
Comment (splatterxl) [inline init.js:24]: unnecessary

File: init.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+function includeHTML() {
+	var z, i, elmnt, file, xhttp;
```
Comment (splatterxl) [inline init.js:2]: use verbose names and don't use var

**Review by splatterxl — CHANGES_REQUESTED** [review]

File: about/index.html (hunk 2 above)
Comment (Oakchris1955) [reply about/index.html:39]: I tried, it wouldn't tho

File: about/userinfo.json (hunk 3 above)
Comment (Oakchris1955) [reply about/userinfo.json:1]: It is JSON, lol

File: init.js (hunk 5 above)
Comment (Oakchris1955) [reply init.js:1]: Who uses old browsers?

File: init.js (hunk 6 above)
Comment (Oakchris1955) [reply init.js:11]: why? This works good as well

File: init.js (hunk 8 above)
Comment (Oakchris1955) [reply init.js:18]: Cuz I made this code like 2 years ago first time and I had to, else it won't work

File: init.js (hunk 10 above)
Comment (Oakchris1955) [reply init.js:2]: Don't you want old browser compatibility?

File: init.js (hunk 9 above)
Comment (Oakchris1955) [reply init.js:24]: oh, alright, lemme change this

Comment (Oakchris1955) [reply init.js]: Fixed

File: init.js (hunk 10 above)
Comment (Oakchris1955) [reply init.js]: Resolved

File: general/bar.html (hunk 4 above)
Comment (Oakchris1955) [reply general/bar.html:1]: alright, but approve it @splatterxl

Comment (Oakchris1955) [reply general/bar.html]: Resolved

File: about/userinfo.json (hunk 3 above)
Comment (splatterxl) [reply about/userinfo.json:1]: wdym

File: init.js (hunk 5 above)
Comment (splatterxl) [reply init.js:1]: a lot of people

File: init.js (hunk 6 above)
Comment (splatterxl) [reply init.js:11]: no, it doesn't

File: about/userinfo.json (hunk 3 above)
Comment (Oakchris1955) [reply about/userinfo.json:1]: doesn't matter

File: init.js (hunk 5 above)
Comment (Oakchris1955) [reply init.js:1]: Make this yourself

File: init.js (hunk 6 above)
Comment (Oakchris1955) [reply init.js:11]: fix it yourself instead of complaining

**Review by moritz-t-w — COMMENTED** [review]
It seems like a nice way of making our code less repetitive, tho it's a little rough around the edges. I will take a closer look later.

**Comment by splatterxl** [comment]
@Oakchris1955 this is _your_ PR. I'm not going to fix your code for you

**Comment by Oakchris1955** [comment]
> It seems like a nice way of making our code less repetitive, tho it's a little rough around the edges. I will take a closer look later.

That is the whole point of this

**Comment by splatterxl** [comment]
Updated for much better code style, haven't tested yet but should work fine

**Comment by splatterxl** [comment]
okay wow this is broken

**Comment by moritz-t-w** [comment]
> > It seems like a nice way of making our code less repetitive, tho it's a little rough around the edges. I will take a closer look later.
> 
> That is the whole point of this

yeah... I wasn't telling you that, I was just commenting that it's a nice way of achieving that point

**Comment by splatterxl** [comment]
Alright folks! I've updated the script that loads the JS files as well as reorganising the entire repository.

**New changes**:
- assets should go in `/assets`, in their respective folder (this includes JS/CSS files but _not_ HTML)
- the user info JSON file has been moved to `/assets`, and the new format is as follows:

```json
{
	"username": "username/alias",
	"name": "previously firstName",
	"icon": "github:[github ID]",
	"alt": "alt text for icon",
	"bio": "text about yourself",
	"connections": [
		{
			"type": "github",
			"name": "your github slug"
		},
		{
			"type": "discord",
			"name": "discord tag (not ID)"
		}
	]
}
```

**Comment by Oakchris1955** [comment]
> okay wow this is broken

Just fixed it with afa5f427b029ba6bbc3de55677338ccd484b7058

File: assets/about.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
+{
+	"users": [
+		{
+			"username": "Oakchris1955",
```
Comment (LukasDoesDev) [inline assets/about.json:4]: gay json

**Review by LukasDoesDev — CHANGES_REQUESTED** [review]

**Review by Oakchris1955 — COMMENTED** [review]
I'll bring those back once you are done

File: assets/about.json [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+                },
+                {
+                    "platform": "discord",
+                    "handle": "Lukas#5427"
+                }
+            ]
+        },
+        {
+            "names": {
+                "displayName": "SplatterXL",
```
Comment (splatterxl) [inline assets/about.json:47]:
```suggestion
                "displayName": "Splatter",
```

File: assets/about.json [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+                "aliases": [
+                    "splatterxl",
+                    "splat",
+                    "spl&aumlt"
```
Comment (splatterxl) [inline assets/about.json:51]:
```suggestion
                    "spl&auml;t"
```

File: assets/js/load_about.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+		members
+	}));
+
+	function createConnection(platform, handle) {
+		const img = document.createElement('img');
+		img.src = '/assets/img/' + platform + '.svg';
+		img.alt = `${platform} icon`;
+
+		const elem = document.createElement("a");
+		switch (platform) {
```
Comment (splatterxl) [inline assets/js/load_about.js:23]: add gitlab here

**Review by splatterxl — DISMISSED** [review]

File: assets/about.json [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+            "names": {
+                "displayName": "Splatter",
+                "aliases": [
+                    "splatterxl",
+                    "splat",
+                    "spl&auml;t"
+                ]
+            },
+            "avatar": "github:74307083",
+            "bio": "Originally a Discord bot developer, now I create small Rust applications.",
```
Comment (splatterxl) [inline assets/about.json:55]:
```suggestion
            "bio": "Avid learner and Rust, TypeScript, Elixir developer.",
```

File: assets/js/load_about.js (hunk 14 above)
Comment (Oakchris1955) [reply assets/js/load_about.js:23]: wdym

**Comment by splatterxl** [comment]
@moritz-t-w can you add gitlab & instagram to the `createConnection` function?

**Comment by splatterxl** [comment]
this needs a rebase

**Comment by moritz-t-w** [comment]
> @moritz-t-w can you add gitlab & instagram to the `createConnection` function?

Will do but later

**Review by moritz-t-w — APPROVED** [review]
Good to go

File: assets/js/load_about.js [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+					setTimeout(() => {
+						elem.textContent = undefined;
+						elem.appendChild(img);
+					}, 2000);
+				};
+				break;
+			}
+			case 'gitlab': {
+				elem.href = `https://gitlab.com/${handle}`;
+				elem.target = 'blank';
```
Comment (splatterxl) [inline assets/js/load_about.js:44]:
```suggestion
				elem.target = '_blank';
```

File: assets/js/load_about.js [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+				break;
+			}
+			case 'instagram': {
+				elem.href = `https://instagram.com/${handle}`;
+				elem.target = 'blank';
```
Comment (splatterxl) [inline assets/js/load_about.js:49]:
```suggestion
				elem.target = '_blank';
```

**Review by splatterxl — APPROVED** [review]

**Pull request merged**